lib/msal-core/src/UserAgentApplication.ts

```diff
@@ -479,13 +475,8 @@ export class UserAgentApplication {
     // If no session exists, prompt the user to login.
     const scope = scopes.join(" ").toLowerCase();
     if (!userObject && !(extraQueryParameters && (extraQueryParameters.indexOf(Constants.login_hint) !== -1 ))) {
-    // TODO: This should be replaced with error callback
-    // TODO: Is this always accessToken?
-      if (this._tokenReceivedCallback) {
-        this._logger.info("User login is required");
-        this._tokenReceivedCallback(ErrorDescription.userLoginError, null, ErrorCodes.userLoginError, Constants.accessToken, this.getUserState(this._cacheStorage.getItem(Constants.stateLogin, this.storeAuthStateInCookie)));
-        return;
-      }
+      this._logger.info("User login is required");
```

Comment:
ideally this log statement should move to the ClientAuthError class (where you create different types of errors)

Reply:
We are tracking an item for telemetry, we will revisit the logging when we come to this item.